File: Oving1/oving1.py
from threading import Thread

i=0

def thread0():
	global i
	for x in range (1000000):
		i+=1
	
def thread1():
	global i
	for x in range(1000000):
		i-=1

def main():
	inc_i_thread = Thread(target=thread0, args = (), )
	inc_i_thread.start()

	dec_i_thread = Thread(target=thread1, args = (), )
	dec_i_thread.start()

	inc_i_thread.join()
	dec_i_thread.join()

	print(i)

main()

File: Oving1/oving1.c
#include <pthread.h>
#include <stdio.h>
 int i = 0 ;

void* thread0(){

	for(int counter=0; counter<1000000; counter++){
		i++;	
	}	
	return NULL;

};

void* thread1(){

	for(int counter=0; counter<1000000; counter++){
		i--;	
	}
return NULL;
};	

int main(){
	pthread_t inc_i_thread;
	pthread_t dec_i_thread;
	pthread_create(&inc_i_thread, NULL, thread0, NULL);
	pthread_create(&dec_i_thread, NULL, thread1, NULL);
	pthread_join(inc_i_thread,NULL);
	pthread_join(dec_i_thread,NULL);
	printf("%d",i);
	printf("\n");

	return 0;
};

